### PR TITLE
Rewrite atop mina

### PR DESCRIPTION
**Pull request by djspiewak**
Apache MINA provides an *asynchronous* SSH library including both client and server functionality. The API is more complex but the functionality is vastly better than SSHJ, which solely exposes a synchronous interface.

### CODE REVIEW

**Comment by djspiewak** [comment]
@wemrysi This should be ready for a real look now!

File: core/src/main/scala/fs2/io/ssh/Client.scala [hunk 1]
```diff
@@ -17,85 +17,112 @@
...
-      r.map(new Process[F](_, blocker, chunkSize))
+      success <- Resource.liftF(fromFuture(F.delay(session.auth())))
+
+      // TODO handle auth failure (minor problems...)
+
+      channel <- Resource.make(
+        for {
+          channel <- F.delay(session.createExecChannel(command))
+          _ <- F.delay(channel.setStreaming(ClientChannel.Streaming.Async))
+          opened <- fromFuture(F.delay(channel.open()))
```
Comment (wemrysi) [inline core/src/main/scala/fs2/io/ssh/Client.scala:99]: Does the command begin executing here? If so, might we miss data from the in/out/err streams? Maybe we can push the `open` into the in/out/err streams on the returned `Process` using an `MVar` as a latch to coordinate?

File: core/src/main/scala/fs2/io/ssh/MinaFuture.scala [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import org.apache.sshd.common.io.{IoReadFuture, IoWriteFuture}
+import org.apache.sshd.client.session.ClientSession
+import org.apache.sshd.client.future.{AuthFuture, ConnectFuture}
+
+import scala.{Array, Boolean, Byte, Unit}
+import scala.util.{Left, Right}
+
+import java.lang.{SuppressWarnings, Throwable}
+
+// this is an unsafe abstraction which exists solely because it should already exist
```
Comment (wemrysi) [inline core/src/main/scala/fs2/io/ssh/MinaFuture.scala:36]: Stunning.

**Review by wemrysi — APPROVED** [review]
Ah the lengths we go for async :wink: 

Had a question around race-condtions on exec channel open and the related IO streams, but otherwise :+1:

File: core/src/main/scala/fs2/io/ssh/Client.scala (hunk 1 above)
Comment (djspiewak) [reply core/src/main/scala/fs2/io/ssh/Client.scala:99]: I'd love to hop on a quick call and talk to you about this. There seems to be some major witchcraft here inside of sshd, but whether there is or isn't, what we do about it from *our* side gets really complicated without compromising resource safety.

**Pull request merged**